PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetPerformance.cxx

```diff
@@ -2304,21 +2376,27 @@ void AliAnalysisTaskEmcalJetPerformance::ComputeBackground()
     
     // Loop over clusters, if the jet container is for full jets. Sum the cluster ET:
     // (1) in the EMCal, (2) in the EMCal fiducial volume, (3) in the DCal, (4), in the DCal fiducial volume, (5) in the EMCal random cone, (6) in the DCal random cone
-    if (jetCont->GetClusterContainer()) {
-      AliClusterContainer* clusCont = GetClusterContainer(0);
-      
-      AliTLorentzVector clus;
+    TString jetContName = jetCont->GetName();
+    if (!jetContName.Contains("Charged")) {
```

Comment:
This is not what we discussed! This is how I used to do the check, but changed to the above GetClusterContainer since it is more efficient. I think we should not change it, there is no reason it needs to be changed.

Reply:
This is not what we discussed since I double checked and there is **at the moment** no getter for the jetType. Markus is looking into it and I can update it once it is added to the container.

This what is changed is NOT what you did before. This check gets a fundamental property of the jet that must not be given to it in a secondary way. It is with it since its creation. The name is created by the type. If type is full jets the name contains "Full" if the type is charged jets the name contains "Charged". Just like it contains the radius in the name or the jet reconstruction algorithm. It is not more efficient to ask for a cluster container that one has to pass to the jet before. Likewise I could pass an element called clown to the jet and then ask if it contains an element clown, as a fancy way of encoding what kind of jet it is. I can likewise add a cluster container to a charged-jet container. That doesn't make it a full jet!
I am really trying to improve this here so please be a little more supportive!

Reply:
I think you misunderstand me. The check you implement was the check I initially implemented here a while back, but it was suggested and discussed to change to what it currently is since it is more efficient not to do the string comparison so often. Thinking about it again, I don't really care that much. I was surprised to see this change because we discussed a bunch yesterday and then you did something different.

In general it would be good if you open separate PR for new features compared to edits you want to make to the task. I try to be quick to review your new features so you can keep working (I am on vacation as you know), but for changes it's better to have them separate so we can slowly discuss. I know you want to make a lot of improvements, but I am very hesitant to change things since the analysis is nearly published and there is risk to doing these changes (we do not implement any tests to make sure behavior doesn't change -- if you really want to make a lot of changes the ideal thing would be to first implement tests). 

Anyway, I am fine now if you want to make the change after all, I don't have a preference.
But please in future make separate PR for changes to the current code. 
And please look at the diff more carefully in GitHub before the PR.